File: ButterKnife/app/src/main/java/com/example/butterknife/MainActivity.java
package com.example.butterknife;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import java.lang.reflect.InvocationTargetException;

@BindLayoutId(R.layout.activity_main)
public class MainActivity extends AppCompatActivity {
    @MyBindView(R.id.button1)
    Button button1;
    @MyBindView(R.id.button2)
    Button button2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try {
            MyButterKnifeUtil.injectLayoutId(this);
            MyButterKnifeUtil.injectViewId(this);
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        button1.setText("button1 - mybindview");
        button2.setText("button2 - mybindview");
    }
}

File: ViewModel/app/src/main/java/com/example/viewmodel/MainActivity.kt
package com.example.viewmodel

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var myViewModel: MyViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        myViewModel = ViewModelProviders.of(this).get(MyViewModel::class.java)
        myViewModel.getLikedNumber().observe(this, object : Observer<Int> {
            override fun onChanged(t: Int?) {
                text_view.setText("$t")
            }
        })
        button1.setOnClickListener {
            myViewModel.addLikedNumber(1)
        }
        button2.setOnClickListener {
            myViewModel.addLikedNumber(-1)
            val intent = Intent()
            intent.setClass(this, MainActivity2::class.java)
            startActivity(intent)
        }
    }
}